### PR TITLE
Rick/vm 3190 vuln details with fields

### PR DESCRIPTION
**Pull request by rick-kenna**
## VM-3190

Need a code example using selected fields and details.

### CODE REVIEW

**Review by jaredkalmus — CHANGES_REQUESTED** [review]
I'm still seeing the same error as last time

`File: vulns_2924781.jsonl with 4607 vulns.
Traceback (most recent call last):
  File "C:\Users\jaredk\OneDrive - Cisco\Documents\Scripts\vuln_details_export\blog_get_vuln_details.py", line 333, in <module>
    for line_num, vuln_line in enumerate(jsonl_f):
  File "C:\Users\jaredk\Miniconda3\envs\pandas_env\lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 7268: character maps to <undefined>`

**Review by kennamatt — APPROVED** [review]
It meets a minimum--small changes would probably make things a bit more tidy and if we can solve what's Jared's problem is and document around that before we unleash it on more people--that would be wise

**Comment by jaredkalmus** [comment]
> It meets a minimum--small changes would probably make things a bit more tidy and if we can solve what's Jared's problem is and document around that before we unleash it on more people--that would be wise

Even if/when my issue gets resolved we should still put a disclaimer in the readme that data from scanner details is inherently "dirty" and customer may experience errors depending on the raw details provided by the scanner. Otherwise we'll be stuck in a pit trying to sanitize all kinds of crazy stuff.

**Comment by kennamatt** [comment]
Probably.  Part of me gets extremely bothered when there's garbage floating around anywhere in our system because you can't really address where / when it will surface.  If we're lucky its ugly, if we're unlucky--we get this and things fall apart.  If we know its garbage in/garbage out--the we probably need to get more clever with the garbage in--we can't just disallow data on automation--but we can strip out / replace stuff that we know is risky.  I don't know if its more an importer or connector thing--but we should have something upstream that makes sure all the text hitting the system is valid utf8.

**Review by jaredkalmus — APPROVED** [review]
Rick was able to successfully complete the script with my API key on his Linux box. I'm cool with chalking up my error to Powershell/Windows/Miniconda weirdness.

File: python/vuln_details/blog_get_vuln_details.py [hunk 1]
```diff
@@ -311,24 +325,30 @@ def convert_to_json(vuln_line):
...
+            
             num_lines += 1
 
     # Close up shop and publish results.
     json_f.close()
+    html_f.write(get_html_tail_stuff())
     html_f.close()
-    print(f"All pau.")
+    
+    print(f"All pau.")   # "pau" in Hawaiian means "done".
```
[the remark this answers is not in the document]
Comment (rick-kenna) [reply python/vuln_details/blog_get_vuln_details.py:351]: You're correct.  In `"Output files:"`, I like it for the alignment.

[the remark this answers is not in the document]
Comment (rick-kenna) [reply python/vuln_details/blog_get_vuln_details.py:351]: Removed formatter prefix in line 351.

File: python/vuln_details/blog_get_vuln_details.py [hunk 2]
```diff
@@ -68,6 +68,18 @@ def convert_from_jsonl(vuln_line):
     
     return json_vuln
 
+def get_html_head_stuff(export_id, selected_fields):
+    line = "<html>\n<header>\n"
+    line += f"<h1>Vulnerability Export ID {export_id}</h1>\n"
+    str_selected_fields = ', '.join(selected_fields)
+    line += f"<p><i><b>with selected fields:</b> {str_selected_fields}</i></p>\n"
+    line += "</header>\n<body>\n"
+    return line
```
[the remark this answers is not in the document]
Comment (rick-kenna) [reply python/vuln_details/blog_get_vuln_details.py:77]: Thanks, much cleaner.

File: python/vuln_details/blog_get_vuln_details.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kennamatt) [reply python/vuln_details/blog_get_vuln_details.py:351]: eh.  You can heredoc this as well, and forget all about alignment--but I get what you're saying--its all lined up.

**Pull request merged**